docs/reference/sql/syntax/lexical-structure.md

```diff
@@ -0,0 +1,201 @@
+SELECT * FROM t1 WHERE k1='foo' EMIT CHANGES;
+```
+
+## Keywords
+
+Some tokens, such as `SELECT`, `INSERT`, and `CREATE`, are _keywords_.
+Keywords are reserved tokens that have a specific meaning in ksqlDB's syntax.
+They control their surrounding allowable tokens and execution semantics.
+Keywords are case insensitive, meaning `SELECT` and `select` are equivalent.
+You can't create an identifier that is already a keyword, unless you use
+backticked identifiers.
+
+A complete list of keywords can be found in the [appendix](../appendix.md#keywords).
+
+## Identifiers
+
+Identifiers are symbols that represent user-space entities, like streams,
+tables, columns, and other objects. For example, if you have a stream named
+`s1`, `s1` is an _identifier_ for that stream. By default, identifiers are
+case-insensitive, meaning `s1` and `S1` refer to the same stream. Under the
+hood, ksqlDB capitalizes all of the characters in the identifier for all
+future display purposes.
+
+Unless an identifier is backticked, it may be composed only of characters that
+are a letter, number, or underscore. There is no imposed limit on the number of
+characters.
+
+To make it possible to use any character in an identifier, you can enclose it
+in backticks (``` ` ```) when you declare and use it. A _backticked identifier_
+is useful when you don't control the data, so it might have special characters,
+or even keywords. When you use backticked identifers, ksqlDB  captures the case 
+exactly, and any future references to the identifer become case-sensitive. For
+example, if you declare the following stream:
+
+```sql
+CREATE STREAM `s1` (
+    k VARCHAR KEY,
+    `@MY-identifier-stream-column!` INT
+) WITH (
+    kafka_topic = 's1',
+    partitions = 3,
+    value_format = 'json'
+);
+```
+
+You must select from it by backticking the stream name and column name and
+using the original casing:
+
+```sql
+SELECT `@MY-identifier-stream-column!` FROM `s3` EMIT CHANGES;
+```
+
+## Constants
+
+There are three implicitly typed constants, or literals, in ksqlDB: strings,
+numbers, and booleans.
+
+### String constants
+
+A string constant is an arbitrary series of characters surrounded by single
+quotes (`'`), like `'Hello world'`. To include a quote inside of a string
+literal, escape the quote by prefixing it with another quote, for example
+`'You can call me ''Stuart'', or Stu.'`
+
+### Numeric constants
+
+Numeric constants are accepted in the following forms:
+
+1. **_`digits`_**
+2. **_`digits`_**`.[`**_`digits`_**`][e[+-]`**_`digits`_**`]`
+3. `[`**_`digits`_**`].`**_`digits`_**`[e[+-]`**_`digits`_**`]`
+4. **_`digits`_**`e[+-]`**_`digits`_**
+
+where **_`digits`_** is one or more single-digit integers (`0` through `9`).
+
+- At least one digit must be present before or after the decimal point, if
+  there is one.
+- At least one digit must follow the exponent symbol `e`, if there is one.
+- Spaces and underscores (nor any other characters) are allowed in the constant.
```

Comment:
Rephrase: No spaces, underscores, or any other characters are allowed in the constant